controllers/horizon_controller.go

```diff
@@ -700,6 +738,18 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 
 	// Create ConfigMaps and Secrets - end
 
+	// Check if keystone service exists for watcher in the same namespace
+	enabledServices := make(map[string]string)
+	for _, service := range keystoneServicesWatch {
+		enabledServices[service] = "no"
+		keystoneService, err := keystonev1.GetKeystoneServiceWithName(ctx, helper, service, instance.Namespace)
+		if err == nil && keystoneService != nil {
+			enabledServices[service] = "yes"
+		}
```

Comment:
The error isn't being handled here though. Maybe something like:

```suggestion
		keystoneService, err := keystonev1.GetKeystoneServiceWithName(ctx, helper, service, instance.Namespace)
		if err != nil  {
			return ctrl.Result{}, err
		}
		
		enabledServices[service] = "no"
		if keystoneService != nil {
		    enabledServices[service] = "yes"
		}
```

Reply:
IIUC, with your proposed patch, if the watcher service is not found the reconcile will return error and keep retrying instead of go ahead and create the deployment. However, in that case we want it to go on with the expected status, so I'd say we should not handle the error case?

My doubt is if my double condition "err == nil && keystoneService != nil" is redundant as maybe just checking for err would be enough.

Reply:
I've implemented this in last version using `if err != nil && !k8s_errors.IsNotFound(err)` . I think that makes what we need it to do.

Reply:
Sure, that sounds good to me. As long as we're checking it and not continuing on and causing a harder to trace failure somewhere further down the line.

Thanks.